lib/cfn-nag/custom_rules/LambdaFunctionCloudWatchLogsRule.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/util/wildcard_patterns'
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class LambdaFunctionCloudWatchLogsRule < BaseRule
+  IAM_CREATEGROUP_PATTERNS = wildcard_patterns('CreateLogGroup').map! { |x| 'logs:' + x } + ['*']
+  IAM_CREATESTREAM_PATTERNS = wildcard_patterns('CreateLogStream').map! { |x| 'logs:' + x } + ['*']
+  IAM_PUTEVENT_PATTERNS = wildcard_patterns('PutLogEvent').map! { |x| 'logs:' + x } + ['*']
+
+  def rule_text
+    'Lambda functions require permission to write CloudWatch Logs'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W55'
+  end
+
+  def audit_impl(cfn_model)
+    # Iterate over each Lambda function
+    lambda_functions = cfn_model.resources_by_type('AWS::Lambda::Function')
+    violating_lambda_functions = lambda_functions.select do |lambda_function|
+      # Determine how the role is assigned to the function
+      # and get the role ID
+      next lambda_function if lambda_function.role.instance_of? String
+
+      # Find role in template if exists
+      role_resources = find_role_by_id?(cfn_model, compute_role_id?(lambda_function.role.values[0]))
```

Comment:
did you try to use the References object from cfn-model here?  it might deal with some of this and keep the rule simpler

Reply:
I looked at it a bit today, however it seems to only handle References (hence the name :) ).  There's a possibility the role could be attached with GetAtt.  I'll look into adding an attribute to the LambdaFunction class which computes the role upon instantiation.  Thanks for the pointer!